Source/Urho3D/Graphics/OpenGL/OGLVertexBuffer.cpp

```diff
@@ -80,6 +80,11 @@ void VertexBuffer::Release()
 
 bool VertexBuffer::SetData(const void* data)
 {
+    if (!vertexCount_)
+    {
+        return true;
```

Comment:
Im late to the party, but this looks fishy. Seems like function lies about what it did. So we pass a null ptr. If we assume it is a correct input then how do we interpret it? Logical conclusion would be "empty data", so we should unset vertex buffer in order to return `true`. This check however implies null ptr is an invalid parameter, but function still "succeeds" while doing nothing. This is odd.

Reply:
Some allocation before that returns nullptr if you try to allocate 0 bytes.

Reply:
So, could it be `if (!vertexCount_ && !data)`?

Reply:
It could. Allocation result of 0 size seems to be compiler dependent. Some may return nullptr, some may return a valid pointer.

Maybe the check should really be before calling this since if there are no vertices then there is nothing to do.

Reply:
It would make sense to return `true` if parameter requests object state change and we are already in that state, so like you said `nullptr` + already no vertices then `true` is good. `nullptr` param with existing vertices maybe should clear vertices and return `true`? unless there is another way of clearing them we should use